milli/src/update/index_documents/extract/extract_geo_points.rs

```diff
@@ -0,0 +1,49 @@
+use std::fs::File;
+use std::io;
+
+use concat_arrays::concat_arrays;
+use log::warn;
+use serde_json::Value;
+
+use super::helpers::{create_writer, writer_into_reader, GrenadParameters};
+use crate::{FieldId, InternalError, Result};
+
+/// Extracts the geographical coordinates contained in each document under the `_geo` field.
+///
+/// Returns the generated grenad reader containing the docid as key associated to the (latitude, longitude)
+pub fn extract_geo_points<R: io::Read>(
+    mut obkv_documents: grenad::Reader<R>,
+    indexer: GrenadParameters,
+    geo_field_id: Option<FieldId>, // faire un grenad vide
+) -> Result<grenad::Reader<File>> {
+    let mut writer = tempfile::tempfile().and_then(|file| {
+        create_writer(indexer.chunk_compression_type, indexer.chunk_compression_level, file)
+    })?;
+
+    // we never encountered any documents with a `_geo` field. We can skip entirely this step
+    let geo_field_id = match geo_field_id {
+        Some(geo) => geo,
+        None => return Ok(writer_into_reader(writer)?),
+    };
+
+    while let Some((docid_bytes, value)) = obkv_documents.next()? {
+        let obkv = obkv::KvReader::new(value);
+        let point = match obkv.get(geo_field_id) {
+            Some(point) => point,
+            None => continue,
+        };
+        let point: Value = serde_json::from_slice(point).map_err(InternalError::SerdeJson)?;
+
+        if let Some((lat, lng)) = point["lat"].as_f64().zip(point["lng"].as_f64()) {
+            // this will create an array of 16 bytes (two 8 bytes floats)
+            let bytes: [u8; 16] = concat_arrays![lat.to_ne_bytes(), lng.to_ne_bytes()];
+            writer.insert(docid_bytes, bytes)?;
+        } else {
+            // TAMO: improve the warn
+            warn!("Malformed `_geo` field");
```

Comment:
we may return an error. For the moment any error in a document stop the indexation

Reply:
Are we sure we want to stop the whole indexation for one malformed document? :thinking: 
@Kerollmops to do a simple warn I think, so I went with it

Reply:
I summon @gmourier for this, but I highly recommend not stop the whole indexation but rather just log that the document with external id `X` has an incorrect `_geo` field format. You can see how I have done this in the `docid_word_positions.rs` where we log the words that are too long (checked with the `valid_mdb_key` function).

Reply:
My POV is: "a transaction is committed only if everything is ok. There is no partially valid transaction".
But in another way, the facets fields that don't have a good format are just ignored without printing any warning. 🤷